pkg/server/file.go

```diff
@@ -130,6 +135,22 @@ func (t *TranslateServiceServer) UploadTranslationFile(
 		return nil, status.Errorf(codes.Internal, "")
 	}
 
+	if messages.Original && len(allMessages) != 0 {
+		// TODO: optimize performance when populateTranslations param is true, currently saveMessages() will be called twice.
```

Comment:
you could change to` t.updateAlteredMessages` and ` t.populateTranslatedMessages` to return an `[]model.Messages`, and then in handler call `SaveMessages`, in one for loop

As another improvement, we could define a new method for repos, `SaveMessagesSlice`, or something like that, which takes an `[]model.Messages` as argument

Reply:
yeah there are a lot of possible improvements, wanted avoid any scope creep.